stdlib/nats/src/main/ballerina/nats/error.bal

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents an I/O error.
+#
+# + message - error message.
+# + id - error id.
+public type IOError record {|
```

Comment:
A bit misleading to call a record an error isn't it?

Reply:
When we return errors we include error details in a record. Typically the convention we follow to include the detail error messages was in the following format which is prefixed by *Error

https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/crypto/src/main/ballerina/crypto/crypto.bal#L114

https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/http/src/main/ballerina/http/http_commons.bal#L178

The record explains the IOError thrown by NATS.